### DISCUSSION TITLE
Signature help

### DISCUSSION THREAD

**Discussion by matoous** [helix-editor/helix#1329] [Ideas] [answered]
Another nice functionality that can be built on top of LSPs is _signature help_. Vim equivalent: https://github.com/ray-x/lsp_signature.nvim. While helpful it also clutters the UI quite a bit, especially when combined with autocomplete. Something worth considering?

**Comment by kirawi** [comment]
Press `space` + `k` for docs of the thing the cursor is hovering over, and #417 adds virtual text. Not sure if that's exactly what you mean.

**Reply by matoous** [reply]
Kind off. Signature help shows just the types expected by given function instead of the whole documentation. Very similar but slightly different.

**Reply by archseer** [reply]
https://microsoft.github.io/language-server-protocol/specification#textDocument_signatureHelp

**Comment by archseer** [comment, marked as answer]
I turned this into an issue with more context https://github.com/helix-editor/helix/issues/1331
